### PR TITLE
Remove deprecated code, old files and use virtualenv

### PR DESCRIPTION
**Pull request by MridulS**
This PR cleans up this repository to make it work with the main branch of astropy. This PR does a bunch of things:

- Remove old files like travis config, parsetab, logs. There is still a cron file which I am not sure if it needs to be there?
- Remove old deprecated API usage.
- Run the repo through ruff to format the python code.
- Use virtualenv to setup envs for running benchmarks.

This PR is required to make https://github.com/astropy/astropy/pull/15779 work.

### CODE REVIEW

**Comment by pllim** [comment]
Since @zacharyburnett and @jhunkeler maintaining the version that STScI is running, maybe would be nice for one of them to take this for a spin first. Otherwise, I welcome the clean-up!

p.s. Though the black stuff makes the diff hard to read.

**Review by pllim — APPROVED** [review]

**Comment by pllim** [comment]
@MridulS , if none of them respond, please ping me again (say, next week) and I can just YOLO-merge this. Thanks!

File: asv.conf.json [hunk 1]
```diff
@@ -3,14 +3,18 @@
...
-}
+  "pythons": ["3.11"],
+  "matrix": {
+    "Cython": [],
+    "build": [],
+    "packaging": [],
+    "scipy": [],
+    "matplotlib": []
+  },
+  "environment_type": "virtualenv"
```
[the remark this answers is not in the document]
Comment (zacharyburnett) [reply asv.conf.json:19]: `environment.yml` could be moved to the root directory, and be whittled down to
```yaml
channels:
  - conda-forge
  - defaults

dependencies:
  - build
  - Cython
  - packaging
  - matplotlib
  - scipy
```

**Review by zacharyburnett — CHANGES_REQUESTED** [review]
this looks good, but I think should use `mamba` instead of `virtualenv`

File: asv.conf.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (MridulS) [reply asv.conf.json:19]: Thanks for the review, interestingly conda was breaking with the latest asv on my local machine with this setup. All the packages here should have wheels available so we wouldn't really need to worry about building them.

I'll have a look again if I can make this work with conda :)

**Comment by MridulS** [comment]
Also are you using the https://github.com/astropy/astropy-benchmarks/blob/main/cron.sh file anywhere in your pipelines?

**Comment by zacharyburnett** [comment]
> Also are you using the https://github.com/astropy/astropy-benchmarks/blob/main/cron.sh file anywhere in your pipelines?

no, I don't think anything is using that

**Review by zacharyburnett — APPROVED** [review]

File: asv.conf.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (zacharyburnett) [reply asv.conf.json:19]: Ok, that sounds good to me

**Pull request merged**